File: Cano-Rodriguez-Claudio/Procesos/Java/fork.java
import java.io.IOException;

public class ForkExample {
    public static void main(String[] args) {
        // Check if the program has the correct number of arguments
        if (args.length != 1) {
            System.err.println("ERR: no params");
            return;
        }

        int N = Integer.parseInt(args[0]); // Convert argument to integer

        // Loop to create N processes
        for (int i = 1; i <= N; i++) {
            try {
                Process process = new ProcessBuilder("java", "-version").start();
                System.out.println("Started Process PID: " + process.pid());
            } catch (IOException e) {
                System.err.println("ERROR: Failed to start process");
            }
        }
    }
}

File: Cano-Rodriguez-Claudio/Procesos/Go/fork.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"strconv"
)

func main() {
	// Check if the program has the correct number of arguments
	if len(os.Args) != 2 {
		fmt.Println("ERR: no params")
		return
	}

	N, err := strconv.Atoi(os.Args[1]) // Convert argument to integer
	if err != nil {
		fmt.Println("ERROR: Invalid number")
		return
	}

	// Loop to create N processes
	for i := 1; i <= N; i++ {
		cmd := exec.Command("sleep", "1") // Example command to simulate a process
		err := cmd.Start()
		if err != nil {
			fmt.Println("ERROR: Failed to start process")
		} else {
			fmt.Printf("Started Process PID: %d\n", cmd.Process.Pid)
		}
	}
}
